src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/MigrateApiAction.java

```diff
@@ -0,0 +1,246 @@
+import com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.ConfigV7;
+import com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.InternalUserV7;
+import com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.RoleMappingsV7;
+import com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.RoleV7;
+import com.amazon.opendistroforelasticsearch.security.securityconf.impl.v7.TenantV7;
+import com.amazon.opendistroforelasticsearch.security.ssl.transport.PrincipalExtractor;
+
+public class MigrateApiAction extends AbstractApiAction {
+
+    @Inject
+    public MigrateApiAction(final Settings settings, final Path configPath, final RestController controller, final Client client,
+                            final AdminDNs adminDNs, final ConfigurationRepository cl, final ClusterService cs, final PrincipalExtractor principalExtractor,
+                            final PrivilegesEvaluator evaluator, ThreadPool threadPool, AuditLog auditLog) {
+        super(settings, configPath, controller, client, adminDNs, cl, cs, principalExtractor, evaluator, threadPool, auditLog);
+    }
+
+    @Override
+    protected void registerHandlers(RestController controller, Settings settings) {
+        controller.registerHandler(Method.POST, "/_opendistro/_security/api/migrate", this);
+    }
+
+    @Override
+    protected Endpoint getEndpoint() {
+        return Endpoint.MIGRATE;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    protected void handlePost(RestChannel channel, RestRequest request, Client client, final JsonNode content) throws IOException {
+
+        final Version oldestNodeVersion = cs.state().getNodes().getMinNodeVersion();
+
+        if(oldestNodeVersion.before(Version.V_7_0_0)) {
+            badRequestResponse(channel, "Can not migrate configuration because cluster is not fully migrated.");
+            return;
+        }
+
+        final SecurityDynamicConfiguration<?> loadedConfig = load(CType.CONFIG, true);
+
+        if (loadedConfig.getVersion() != 1) {
+            badRequestResponse(channel, "Can not migrate configuration because it was already migrated.");
+            return;
+        }
+
+        final SecurityDynamicConfiguration<ConfigV6> configV6 = (SecurityDynamicConfiguration<ConfigV6>) loadedConfig;
+        final SecurityDynamicConfiguration<ActionGroupsV6> actionGroupsV6 = (SecurityDynamicConfiguration<ActionGroupsV6>) load(CType.ACTIONGROUPS, true);
+        final SecurityDynamicConfiguration<InternalUserV6> internalUsersV6 = (SecurityDynamicConfiguration<InternalUserV6>) load(CType.INTERNALUSERS, true);
+        final SecurityDynamicConfiguration<RoleV6> rolesV6 = (SecurityDynamicConfiguration<RoleV6>) load(CType.ROLES, true);
+        final SecurityDynamicConfiguration<RoleMappingsV6> rolesmappingV6 = (SecurityDynamicConfiguration<RoleMappingsV6>) load(CType.ROLESMAPPING, true);
+
```

Comment:
Need to add NodesDn also to the migration

Reply:
added.

Reply:
Need to add to validate api too ?